docs/subnets/create-a-simple-rust-vm.md

```diff
@@ -0,0 +1,1256 @@
+    _upgrade_bytes: &[u8],
+    _config_bytes: &[u8],
+    to_engine: Sender<subnet::rpc::common::message::Message>,
+    _fxs: &[subnet::rpc::common::vm::Fx],
+    app_sender: Box<dyn subnet::rpc::common::appsender::AppSender + Send + Sync>,
+) -> io::Result<()> {
+    log::info!("initializing Vm");
+    let mut vm_state = self.state.write().await;
+
+    vm_state.ctx = ctx;
+
+    let version =
+        Version::parse(VERSION).map_err(|e| Error::new(ErrorKind::Other, e.to_string()))?;
+    vm_state.version = version;
+
+    let genesis = Genesis::from_slice(genesis_bytes)?;
+    vm_state.genesis = genesis;
+
+    let current = db_manager.current().await?;
+    let state = state::State {
+        db: Arc::new(RwLock::new(current.db)),
+        verified_blocks: Arc::new(RwLock::new(HashMap::new())),
+    };
+    vm_state.state = Some(state.clone());
+
+    vm_state.to_engine = Some(to_engine);
+
+    self.app_sender = Some(app_sender);
+
+    let has_last_accepted = state.has_last_accepted_block().await?;
+    if has_last_accepted {
+        let last_accepted_blk_id = state.get_last_accepted_block_id().await?;
+        vm_state.preferred = last_accepted_blk_id;
+        log::info!("initialized Vm with last accepted block {last_accepted_blk_id}");
+    } else {
+        let mut genesis_block = Block::new(
+            ids::Id::empty(),
+            0,
+            0,
+            vm_state.genesis.data.as_bytes().to_vec(),
+            choices::status::Status::default(),
+        )?;
+        genesis_block.set_state(state.clone());
+        genesis_block.accept().await?;
+
+        let genesis_blk_id = genesis_block.id();
+        vm_state.preferred = genesis_blk_id;
+        log::info!("initialized Vm with genesis block {genesis_blk_id}");
+    }
+
+    self.mempool = Arc::new(RwLock::new(VecDeque::with_capacity(100)));
+
+    log::info!("successfully initialized Vm");
+    Ok(())
+}
+```
+
+#### create_handlers
+
+Registered handlers defined in `api::chain_handlers::Service`. See [below](create-a-simple-rust-vm.md#api) for more on APIs.
+
+```rust title="/timestampvm/src/vm/mod.rs"
+/// Creates VM-specific handlers.
+async fn create_handlers(
+    &mut self,
+) -> io::Result<HashMap<String, subnet::rpc::common::http_handler::HttpHandler>> {
+    let svc = api::chain_handlers::Service::new(self.clone());
+    let mut handler = jsonrpc_core::IoHandler::new();
+    handler.extend_with(api::chain_handlers::Rpc::to_delegate(svc));
+
+    let http_handler = subnet::rpc::common::http_handler::HttpHandler::new_from_u8(0, handler)
+        .map_err(|_| Error::from(ErrorKind::InvalidData))?;
+
+    let mut handlers = HashMap::new();
+    handlers.insert("/rpc".to_string(), http_handler);
+    Ok(handlers)
+}
+```
+
+#### create_static_handlers
+
+Registered handlers defined in `api::chain_handlers::Service`. See [below](create-a-simple-rust-vm.md#api) for more on APIs.
+
+Note: If this method is called, no other method will be called on this VM.
```

Comment:
You could use https://docusaurus.io/docs/next/markdown-features/admonitions for this Note.